cyberbrain/tracer.py

```diff
@@ -81,6 +82,8 @@ def _initialize_frame_and_logger(
             frame=self.frame,
             debug_mode=self.debug_mode,
         )
+        if self.func_lineno:
+            self.frame.defined_lineno = self.func_lineno
```

Comment:
Don't put `func_lineno` in Tracer. You can just `frame.defined_lineno`

Reply:
Btw, the tests failed because all the tests share the same tracer, thus they also shared the same `func_lineno` (which should be different)

Reply:
Thanks for the pointers. I opted to create a class to handle the traceback. It seems to be working way better than the previous approach. I just need to figure out why the python 3.7 tests are failing.